rules/DeadCode/Rector/ConstFetch/RemovePhpVersionIdCheckRector.php

```diff
@@ -247,26 +253,28 @@ private function processGreater(ConstFetch $constFetch, Greater $greater, If_ $i
     }
 
     /**
-     * @return Stmt[]|null
+     * @return null|If_|Stmt[]
      */
-    private function processGreaterLeft(Greater $greater, If_ $if): ?array
+    private function processGreaterLeft(Greater $greater, If_ $if): null|If_|array
     {
         $value = $greater->right;
         if (! $value instanceof LNumber) {
             return null;
         }
 
-        if ($this->phpVersion >= $value->value) {
-            return $if->stmts;
+        if ($this->phpVersion < $value->value) {
+            return null;
         }
 
-        return null;
+        if ($if->stmts === []) {
+            $this->removeNode($if);
+            return $if;
```

Comment:
@TomasVotruba return `$if` is needed to avoid "applied rules" not shown in the list, ref 

https://github.com/rectorphp/rector-src/blob/2dd8c6b81436f32ff91cabeb6ca658d89a7cae47/src/Rector/AbstractRector.php#L207-L210

which it only shows when it has `node` or `array of stmts` returns:

https://github.com/rectorphp/rector-src/blob/2dd8c6b81436f32ff91cabeb6ca658d89a7cae47/src/Rector/AbstractRector.php#L220-L222

Reply:
The changed should be included for removed nodes too. The `AbstractRector` should be updated instead.

Reply:
Ok, I created PR https://github.com/rectorphp/rector-src/pull/2261 for it.

Reply:
@TomasVotruba that seems not doable by only move up the `RectorWithLineChange` as the information of Node removed is against original node, which if it returns null, it fallback to `clone $node`, which the information is for object before refactor, ref

https://github.com/rectorphp/rector-src/blob/fb8260c3a171ae29ac3a7c01e0dcb91ffb1077a0/src/Rector/AbstractRector.php#L192

it also cause errors in other e2e tests by move it up, ref PR https://github.com/rectorphp/rector-src/pull/2261#issuecomment-1120264389